service/matching/tasklist/task_list_manager.go

```diff
@@ -274,6 +274,10 @@ func (c *taskListManagerImpl) Start() error {
 	if c.taskListID.IsRoot() && c.taskListKind == types.TaskListKindNormal {
 		c.partitionConfig = c.db.PartitionConfig().ToInternalType()
 		c.logger.Info("get task list partition config from db", tag.Dynamic("root-partition", c.taskListID.GetRoot()), tag.Dynamic("task-list-partition-config", c.partitionConfig))
+		if c.partitionConfig != nil {
+			// push update notification to all non-root partitions on start
+			c.notifyPartitionConfig(context.Background(), *c.partitionConfig, int(c.partitionConfig.NumReadPartitions))
```

Comment:
I wonder if we should do this async. Since we don't act on failed-to-notify cases no need to block startup of root partition while it notifies N partitions

Reply:
updated!